src/common/loader/index.js

```diff
@@ -57,23 +57,47 @@ function restArWorker(src) {
     });
 }
 
-function loader(srcList, typeList) {
+function getPromise(typeList, promiseKey, src, key) {
+    if (typeList[key]) {
+        if (typeList[key] === 'css') {
+            return cssLoad(src, promiseKey);
+        } else if (typeList[key] === 'restar-worker') {
+            return restArWorker(src, promiseKey);
+        } else if (typeList[key] === 'module') {
+            return srcLoad(src, promiseKey, 'module');
+        }
+    }
+    return srcLoad(src, promiseKey);
+}
+
+/**
+ * Loader, which loads a script or stylesheet. Will cache the results on the page so it does not load more than once.
+ * @param {*} srcList Array of sources to load
+ * @param {*} typeList The type of each src. Each index should match with srcList. See types for more info
+ * @param {boolean} stagger If true, will wait for one source to load before loading the next.
+ * @returns Promise, when resolved, all sources are loaded
+ *
+ * Types for typelist
+ * src - standard javascript source (default value)
+ * restar-worker - (deprecated), loads restar service worker
+ * module - loads javascript script with type=module
+ * css - CSS source file
+ */
+function loader(srcList, typeList, stagger) {
     const promiseKey = srcList.join(',');
     if (!cachePromises[promiseKey]) {
-        cachePromises[promiseKey] = Promise.all(
-            srcList.map((src, key) => {
-                if (typeList[key]) {
-                    if (typeList[key] === 'css') {
-                        return cssLoad(src, promiseKey);
-                    } else if (typeList[key] === 'restar-worker') {
-                        return restArWorker(src, promiseKey);
-                    } else if (typeList[key] === 'model-viewer') {
-                        return srcLoad(src, promiseKey, 'module');
-                    }
-                }
-                return srcLoad(src, promiseKey);
-            })
-        );
+        if (stagger) {
```

Comment:
I like this idea of staggering. I think it could be optimized a bit more by identifying root flies that don't have dependencies and allow them to load asynchronously. That would speed things up more. Maybe that could be a future optimization?

Reply:
Yes, that's the plan. 
I think ideally we need to change loading to take an object which should have:
name - file to load from cdn
path - the path to the cdn
type - file type (css, script)
dependencies - array of all script that need to laod first. 

But this is for future.